File: Java/src/main/java/Solution901_1000/Solution922.java
package Solution901_1000;

import java.util.ArrayList;

/**
 * @author windows
 */
public class Solution922 {
//    public int[] sortArrayByParityII(int[] nums) {
//        for (int i = 0; i < nums.length; i++) {
//            if (i % 2 == 0 && nums[i] % 2 != 0){
//                for (int j = i + 1; j < nums.length; j++) {
//                    if (j % 2 != 0 && nums[j] % 2 == 0){
//                        int temp = nums[j];
//                        nums[j] = nums[i];
//                        nums[i] = temp;
//                        break;
//                    }
//                }
//            }else if(i % 2 != 0 && nums[i] % 2 == 0){
//                for (int j = i + 1; j < nums.length; j++) {
//                    if (j % 2 == 0 && nums[j] % 2 != 0){
//                        int temp = nums[j];
//                        nums[j] = nums[i];
//                        nums[i] = temp;
//                        break;
//                    }
//                }
//            }
//        }
//        return nums;
//    }

    public int[] sortArrayByParityII(int[] nums) {
        ArrayList<Integer> odd = new ArrayList<>();
        ArrayList<Integer> even = new ArrayList<Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (i % 2 == 0 && nums[i] % 2 != 0){
                odd.add(i);
            }else if(i % 2 != 0 && nums[i] % 2 == 0){
                even.add(i);
            }
        }
        for (int i = 0; i < odd.size(); i++) {
            int indexOdd = odd.get(i);
            int indexEven = even.get(i);
            int temp = nums[indexEven];
            nums[indexEven] = nums[indexOdd];
            nums[indexOdd] = temp;
        }
        return nums;
    }


    public static void main(String[] args) {
        Solution922 solution922 = new Solution922();
    }
}

File: C++/src/Solution901_1000/Solution922.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution922.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

vector<int> sortArrayByParityII(vector<int>& nums) {
  int n = nums.size();
  for (int i = 0; i < n; ++i) {
    if (i % 2 == 0 && nums[i] % 2 != 0){
      for (int j = i + 1; j < n; ++j) {
        if (nums[j] % 2 == 0) {
          swap(nums[i], nums[j]);
          break;
        }
      }
    }else if (i % 2 != 0 && nums[i] % 2 == 0){
      for (int j = i + 1; j < n; ++j) {
        if (nums[j] % 2 != 0) {
          swap(nums[i], nums[j]);
          break;
        }
      }
    }
  }
  return nums;
}
